spring-cloud-generator/showcase/showcase-spring-starter/src/test/java/com/google/showcase/v1beta1/spring/EchoAutoConfigurationTests.java

```diff
@@ -196,8 +196,8 @@ void testCustomTransportChannelProviderUsedWhenProvided() throws IOException {
     // TransportChannelProvider bean
     when(mockApiCallContext.withCredentials(any())).thenReturn(mockApiCallContext);
     when(mockApiCallContext.withTransportChannel(any())).thenReturn(mockApiCallContext);
-    when(mockApiCallContext.withStreamWaitTimeout(any())).thenReturn(mockApiCallContext);
```

Comment:
Is there going to be errors if we don't mock the behaviors of `ApiCallContext`? I guess somewhere in the client initialization logic these lines are called?

Reply:
Yes, removing will cause errors (the same ones we observe in the ci before this fix).
However, for the purpose of this test, instead of creating the `mockTransportChannelProvider`, I think it is also okay to create a real channelProvider and test it is actually used. The benefit is we free ourselves from implementation detail changes in the future.

E.g.
```
      InstantiatingGrpcChannelProvider channelProvider = InstantiatingGrpcChannelProvider
              .newBuilder()
              .setEndpoint("localhost:7469")
              .setMaxInboundMessageSize(Integer.MAX_VALUE)
              .build();
```